api/database/mongooseIdPlugin.js

```diff
@@ -0,0 +1,7 @@
+export default function mongooseIdPlugin(schema, options) {
+  schema.options.toJSON.transform = function (doc, ret, options) {
+    delete ret._id; // Remove internal _id property before exposing it as JSON.
```

Comment:
> But generally, I think we should better avoid having transformations in and out of the database, since it could create "mapping issues".

Totally agree with that.

But. Regarding your first sentence, I want to clarify that `update` is based on the id provided. So it is handled manually. If you can, could you take a quick look in that method? I believe is pretty straightforward but is also discussable.

In certain way I believe what I achieved is robust based on the mongodb & mongoose behavior but maybe someone with more experience have a nicer alternative. Thoughts, ideas?